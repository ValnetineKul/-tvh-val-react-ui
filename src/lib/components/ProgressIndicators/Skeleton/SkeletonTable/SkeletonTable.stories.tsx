import { createMeta, createTemplate } from '../../../../story-utils';
import SkeletonTable from './SkeletonTable';

const Template = createTemplate(SkeletonTable);

export const Primitive = Template.bind({});
Primitive.args = {
  cols: 5,
  rows: 3,
};
Primitive.storyName = 'SkeletonTable';

export default createMeta({
  component: SkeletonTable,
  title: 'Components/ProgressIndicators/Skeleton/SkeletonTable',
});
