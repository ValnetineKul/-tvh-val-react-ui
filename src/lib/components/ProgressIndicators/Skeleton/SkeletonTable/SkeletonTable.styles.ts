import type { Surface } from '../../../../../themes/core';
import { makeStyles } from '../../../../../themes/core';
import { skeletonAnimationStyles } from '../helper';

type Props = {
  onSurface: Surface;
};

const useStyles = makeStyles<Props>({
  name: 'SkeletonTable',
})((theme, { onSurface }) => ({
  tableHead: {
    ...skeletonAnimationStyles(theme.color['bg/surface300/default'], '#A0A0A0'),
    width: '100%',
  },
  tableBody: {
    ...skeletonAnimationStyles(
      onSurface === '200' ? theme.color['bg/surface300/default'] : theme.color['bg/surface200/default'],
      onSurface === '200' ? theme.color['bg/surface400/default'] : theme.color['bg/surface150/default']
    ),
    width: '100%',
  },
  narrowColumn: {
    width: '56px',
  },
}));

export default useStyles;
