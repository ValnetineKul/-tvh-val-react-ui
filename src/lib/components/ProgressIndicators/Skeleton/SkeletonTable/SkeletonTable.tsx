import type { FC } from 'react';
import React from 'react';
import { Skeleton } from '@mui/material';
import { useSurface } from '../../../../../themes/core';
import Table from '../../../Tables/Table';
import TableHead from '../../../Tables/TableHead/TableHead';
import TableBody from '../../../Tables/TableBody/TableBody';
import TableRow from '../../../Tables/TableRow/TableRow';
import TableCell from '../../../Tables/TableCell/TableCell';
import useStyles from './SkeletonTable.styles';

export interface SkeletonTableProps {
  rows: number;
  cols: number;
  isFirstColumnNarrow?: boolean;
}

const SkeletonTable: FC<SkeletonTableProps> = ({ rows, cols, isFirstColumnNarrow = false }) => {
  const { color: onSurface } = useSurface();
  const { classes } = useStyles({ onSurface });

  return (
    <div data-testid="skeleton-table">
      <Table>
        <TableHead>
          <TableRow>
            {new Array(cols).fill(null).map((col, i) => (
              <TableCell key={i} className={i === 0 && isFirstColumnNarrow ? classes.narrowColumn : ''}>
                <Skeleton variant="rectangular" className={classes.tableHead} animation="wave" />
              </TableCell>
            ))}
          </TableRow>
        </TableHead>
        <TableBody>
          {new Array(rows).fill(null).map((row, i) => (
            <TableRow key={i}>
              {new Array(cols).fill(null).map((col, j) => (
                <TableCell key={j} className={j === 0 && isFirstColumnNarrow ? classes.narrowColumn : ''}>
                  <Skeleton variant="rectangular" className={classes.tableBody} animation="wave" />
                </TableCell>
              ))}
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </div>
  );
};

export default SkeletonTable;
