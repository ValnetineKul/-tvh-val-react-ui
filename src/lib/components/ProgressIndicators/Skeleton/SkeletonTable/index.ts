import SkeletonTable from './SkeletonTable';

export type { SkeletonTableProps } from './SkeletonTable';
export default SkeletonTable;
