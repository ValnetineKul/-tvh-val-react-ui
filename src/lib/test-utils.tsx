import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
