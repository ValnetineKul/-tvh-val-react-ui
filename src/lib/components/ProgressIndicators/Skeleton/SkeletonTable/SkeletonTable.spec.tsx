import React from 'react';
import SkeletonTable from './SkeletonTable';
import { render } from '../../../../test-utils';

describe('SkeletonTable', () => {
  it('Should render correctly', () => {
    const { container } = render(<SkeletonTable rows={3} cols={5} />);
    expect(container.firstChild).toMatchSnapshot('SkeletonTable');
  });
});
