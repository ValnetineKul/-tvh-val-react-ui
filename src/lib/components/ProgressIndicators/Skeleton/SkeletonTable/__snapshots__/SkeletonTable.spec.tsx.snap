// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonTable Should render correctly: SkeletonTable 1`] = `
<div
  data-testid="skeleton-table"
>
  <div
    class="MuiTableContainer-root mui-rorn0c-MuiTableContainer-root tss-1giqlv2-Surface-root-TableContainer-root"
  >
    <table
      class="MuiTable-root MuiTable-stickyHeader tss-12i0kxp-Table-root mui-1u90bpa-MuiTable-root"
    >
      <thead
        class="tss-1q9inhp-TableHead-root MuiTableHead-root mui-15wwp11-MuiTableHead-root"
      >
        <tr
          class="tss-18gbisn-TableRow-root MuiTableRow-root MuiTableRow-head mui-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root tss-1cg92jv-TableCell-head MuiTableCell-head tss-wq1z2g-TableCell-stickyHeader MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1i7z953-MuiTableCell-root"
            scope="col"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-1qj49ep-SkeletonTable-tableHead mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </th>
          <th
            class="MuiTableCell-root tss-1cg92jv-TableCell-head MuiTableCell-head tss-wq1z2g-TableCell-stickyHeader MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1i7z953-MuiTableCell-root"
            scope="col"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-1qj49ep-SkeletonTable-tableHead mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </th>
          <th
            class="MuiTableCell-root tss-1cg92jv-TableCell-head MuiTableCell-head tss-wq1z2g-TableCell-stickyHeader MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1i7z953-MuiTableCell-root"
            scope="col"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-1qj49ep-SkeletonTable-tableHead mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </th>
          <th
            class="MuiTableCell-root tss-1cg92jv-TableCell-head MuiTableCell-head tss-wq1z2g-TableCell-stickyHeader MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1i7z953-MuiTableCell-root"
            scope="col"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-1qj49ep-SkeletonTable-tableHead mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </th>
          <th
            class="MuiTableCell-root tss-1cg92jv-TableCell-head MuiTableCell-head tss-wq1z2g-TableCell-stickyHeader MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1i7z953-MuiTableCell-root"
            scope="col"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-1qj49ep-SkeletonTable-tableHead mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root mui-apqrd9-MuiTableBody-root"
      >
        <tr
          class="tss-18gbisn-TableRow-root MuiTableRow-root mui-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
        </tr>
        <tr
          class="tss-18gbisn-TableRow-root MuiTableRow-root mui-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
        </tr>
        <tr
          class="tss-18gbisn-TableRow-root MuiTableRow-root mui-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
          <td
            class="MuiTableCell-root tss-5vks2j-TableCell-body MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium tss-3jd96d-TableCell-root mui-1y0jb3z-MuiTableCell-root"
          >
            <div
              class="tss-gfi776-TableCell-innerRoot tss-12-TableCell-innerRoot-ref"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-a3ojbr-SkeletonTable-tableBody mui-1slz293-MuiSkeleton-root"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
